torchmetrics/functional/classification/kldivergence.py

```diff
@@ -0,0 +1,38 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.data import METRIC_EPS
+
+
+def _kld_update(preds: Tensor, target: Tensor):
```

Comment:
function should check that `preds.shape==target.shape`. You can use this function https://github.com/PyTorchLightning/metrics/blob/078671fa0b28d7d0a9c505b077d9acc8ac28a69d/torchmetrics/utilities/checks.py#L23